libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/UpcomingRoadObject.kt

```diff
@@ -5,22 +5,16 @@ package com.mapbox.navigation.base.trip.model.roadobject
  * or start an alert if it has length.
  *
  * @param roadObject road object
- * @param distanceToStart distance to the start of the alert.
- * If the object has a length, and we've passed the start point,
- * **this value will be negative** until we cross the finish point of the objects's geometry.
- * This negative value, together with [RoadObjectGeometry.length]
- * can be used to calculate the distance since the start of an object.
+ * @param distanceToStart distance to the start of the object.
+ * @param distanceInfo provides extra distance details for road objects coming from
+ * Electronic Horizon.
  */
-class UpcomingRoadObject private constructor(
+class UpcomingRoadObject(
```

Comment:
Can this have a public constructor now? We won't be able to add new values here in the future?

Reply:
it's called from `core` module

Reply:
This means that it should keep the builder pattern for backward compatibility.

Reply:
done